rfcs/0000-c-unwind-and-project-group-charter.md

```diff
@@ -0,0 +1,496 @@
+themselves do not contain any destructors or landing pads -- they
+simply need to be "skipped over", in effect.
+
+Moreover, Lucet executes only on a narrow range of platforms:
+
+* XXX details
+
+### mozjpeg: XXX
+
+XXX somebody fill this in XXX
+
+### lua bindings and longjmp
+
+Although longjmp/setjmp are not directly in scope for this group,
+there is an interesting interaction that was uncovered and is worth
+documenting. On Windows targets specifically, longjmp is implemented
+using the same "Structured Exception Handling" (SEH) mechanism that is
+used for C++ exceptions and other sorts of errors. As a result, in
+some versions of Rust at least, we found that attempts to longjmp over
+Rust frames triggered unwinding mechanisms. Note that in general a
+longjmp over a Rust frame that contains destructors is [Undefined
+Behavior] -- however, we would like to support longjmp for Rust frames
+that do not contain destructors. At minimum, this group should take
+the behavior of longjmp into account; if it is easy to do, we may also
+introduce mechanisms or RFCs that help to specify the use of longjmp
+in particular cases.
+
+## Our proposal
+
+This RFC proposes a multi-prong plan to solve this problem:
+
+* First, to modify Rust functions with the "C" abi to abort on unwinding, as proposed
+  in [rust-lang/rust#52652].
+* Second, to introduce the "C unwind" ABI, with virtually all details
+  (but not quite all) left as "to be determined".
+    * The **intent** of this ABI, however, is to permit Rust panics to
+      be "translated" into a "native" unwinding mechanism, though
+      precisely which is dependent on the target platform or other
+      compiler options.
+        * (In practice, this translation is a no-op, since Rust panics
+          use the native mechanism, but this is explicitly not
+          required for the future.)
+    * Folks currently using the "C" ABI to do unwinding can migrate to this new
+      "C unwind" ABI, which captures their intentions. Their code is no more stable
+      than it was, but they are on the right path.
+* Finally, to create a "project group" with the purpose of designing subsequent
+  RFCs to flesh out the details of the "C unwind" ABI
+    * The "project group" term is newly introduced, but it corresponds
+      to a kind of working group -- one with the goal of fleshing out
+      a particular proposal or completing a project.
+    * We aim to specify how "C unwind" works on major platforms
+    * Goal is to enable Rust panics to propagate across native frames
+    * And perhaps to enable native exceptions to propagate across Rust frames
+    * But not to allow catching or throwing native exceptions from Rust code
+
+The first two goals work together. We simul
+The goal of introducing the "C unwind" ABI is simple: it allows
+existing projects to migrate to the newer ABI and continue to work "as
+well as they ever did". In the meantime, we can tighten up the rules
```

Comment:
```suggestion
well as they ever did" or better, since they will avoid certain optimizations
that are incorrect for their use cases.. In the meantime, we can tighten up the rules
```